src/test/java/org/folio/rest/support/HttpClient.java

```diff
@@ -1,46 +1,55 @@
 package org.folio.rest.support;
 
+import static java.lang.String.format;
+import static javax.ws.rs.core.HttpHeaders.ACCEPT;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_TYPE;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.concurrent.CompletableFuture;
+
 import io.vertx.core.Handler;
 import io.vertx.core.Vertx;
 import io.vertx.core.http.HttpClientRequest;
 import io.vertx.core.http.HttpClientResponse;
 import io.vertx.core.json.Json;
 import io.vertx.core.logging.Logger;
 import io.vertx.core.logging.LoggerFactory;
-import java.net.MalformedURLException;
-
-import java.net.URL;
 
 public class HttpClient {
-  private static final Logger log = LoggerFactory.getLogger(HttpClient.class);
+  private static final Logger LOG = LoggerFactory.getLogger(HttpClient.class);
 
   private static final String TENANT_HEADER = "X-Okapi-Tenant";
+  private static final String X_OKAPI_URL = "X-Okapi-Url";
+  private static final String X_OKAPI_URL_TO = "X-Okapi-Url-to";
 
   private final io.vertx.core.http.HttpClient client;
 
   public HttpClient(Vertx vertx) {
     client = vertx.createHttpClient();
   }
 
-  private void stdHeaders(HttpClientRequest request, String url, String tenantId) {
-    URL tmp = null;
+  private void initDefaultHeaders(HttpClientRequest request, String url, String tenantId) {
     try {
-      tmp = new URL(url);
+      initDefaultHeaders(request, new URL(url), tenantId);
     } catch (MalformedURLException ex) {
-      log.warn("Malformed url:" + url + " message: " + ex.getMessage());
+      LOG.info(format("Malformed url: %s, message: %s", url, ex.getMessage()));
     }
-    stdHeaders(request, tmp, tenantId);
   }
 
-  private void stdHeaders(HttpClientRequest request, URL url, String tenantId) {
-    if (tenantId != null) {
+  private void initDefaultHeaders(HttpClientRequest request, URL url, String tenantId) {
+    if (isNotBlank(tenantId)) {
       request.headers().add(TENANT_HEADER, tenantId);
     }
     if (url != null) {
-      request.headers().add("X-Okapi-Url", url.getProtocol() + "://" + url.getHost() + ":" + url.getPort());
-      request.headers().add("X-Okapi-Url-to", url.getProtocol() + "://" + url.getHost() + ":" + url.getPort());
+      String baseUrl = format("%s://%s", url.getProtocol(), url.getAuthority());
+      request.headers().add(X_OKAPI_URL, baseUrl);
```

Comment:
What is the value of a constant when it is used in only a single place?

Reply:
Because I use standard headers from javax.ws.rs.core.HttpHeaders and decided to extract custom headers to the constant for better readability